core/store/src/trie/ops/iter.rs

```diff
@@ -106,13 +109,27 @@ where
             trie_interface,
             prune_condition,
         };
-        iter.descend_into_node(root)?;
+        if let Some(start) = start {
+            iter.seek_nibble_slice(
+                NibbleSlice::new(start.as_ref()),
+                false,
+                AccessOptions::NO_SIDE_EFFECTS,
+            )?;
+        } else {
+            iter.descend_into_node(root, AccessOptions::DEFAULT)?;
+        }
         Ok(iter)
     }
 
+    /// Skips one element in the iteration without recording it.
+    pub fn skip_one(&mut self, expected_key: &[u8]) {
+        self.iter_step();
+        assert_eq!(expected_key, self.key());
+    }
```

Comment:
This is the only part that I'm not really happy about.

We could alternatively move this to the construction of the iterator, but then it would have an non-intuitive exclusive start interval.